### PR TITLE
Added Shorewall(6)-lite.

### PR DESCRIPTION
**Pull request by wvdakker**
Maintainer: Willem van den Akker/ <wvdakker@wilsoft.nl>
Compile tested: (ZyXel P2812-F1, LEDE Reboot)
Run tested: (ZyXel P2812-F1, LEDE Reboot, Install, configuration from admin system)

Description:
The Shoreline Firewall, is high-level tool for configuring Netfilter.

	Shorewall allows for central administration of multiple IPv4 firewalls
	through use of Shorewall lite. The full Shorewall product is installed
	on a central administrative system where compiled Shorewall scripts are
	generated. These scripts are copied to the firewall systems where they
	run under the control of Shorewall-lite.

### CODE REVIEW

File: net/shorewall-core/Makefile [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#
+# Copyright (C) 2008-2017 OpenWrt.org
```
Comment (diizzyy) [inline net/shorewall-core/Makefile]: https://forum.lede-project.org/t/packaging-guidelines/65/2
Please read "Copyright statements in Makefiles"

File: net/shorewall-core/Makefile [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+PKG_NAME:=shorewall-core
+PKG_VERSION:=5.0.15.2
+PKG_DIRECTORY:=5.0
+PKG_RELEASE:=1
+PKG_MAINVERSION:=5.0.15
+
+PKG_SOURCE_URL:=http://www.shorewall.net/pub/shorewall/$(PKG_DIRECTORY)/shorewall-$(PKG_MAINVERSION)/
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
+PKG_MD5SUM:=1912bf34776837dcca85741aeddd262a
```
Comment (diizzyy) [inline net/shorewall-core/Makefile]: Despite it's name please use SHA256 checksum here

Comment (wvdakker) [inline net/shorewall-core/Makefile]: I would have if they are available. Only sha1sums or md5 can be used.

File: net/shorewall-core/Makefile (hunk 1 above)
Comment (wvdakker) [inline net/shorewall-core/Makefile]: The packages were part of BB. So the Makefiles are copied from BB.
The Copyright for OpenWrt will be the same. I added mine for 2017.

Hope to hear from you.

**Comment by lucize** [comment]
@wvdakker can you make a pr for shorewall-full version ?
I'm using it on some complex multi isp (x86) and I installed it manually, some perl dependencies: perlbase-autoloader, perlbase-autouse, perlbase-dynaloader, perlbase-digest, perlbase-findbin, perlbase-getopt, perlbase-hash
regards

**Comment by wvdakker** [comment]
@lucize: I will make this weekend a shorewall (full) package. It will be big because of all the perl dependencies. Not all router can run this full package.

**Comment by lucize** [comment]
thanks,
the space usage for all the shorewall files including conffiles and macro from usr is about 2MB
I don't think size matters on usecases of shorewall

regards

**Comment by wvdakker** [comment]
It takes more time to create an OpenWrt specific full Shorewall implementation. All install and configfiles must be 'OpenWrt' aware. 
It will be a seperate PR so it won't disturb the *lite transistion to OpenWrt.

**Comment by lucize** [comment]
I have several futro s550 that use CF cards, and I use some industrial 128MB cards so I can't install a full system, they cand do ~800MB/s routing, high speed vnp too

regards

**Comment by wvdakker** [comment]
Little tiny ping ping...
Someone willing to pull the request in?

**Comment by lucize** [comment]
is there a reason not using 5.1.4 ?
I've made a full package based on your model, but I'm using the 5.1.4 version, I would like to add it, but it depends on shorewall-core and yours is 5.0.x version
The full version needs:         $(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/shorewall $(1)/usr/sbin found in core archive

File: net/shorewall-lite/files/vardir [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# move state dir out of ram
+VARDIR=/etc/shorewall-lite/state
```
Comment (lucize) [inline net/shorewall-lite/files/vardir]: why not have it in tmpfs like /tmp/state ?

Comment (wvdakker) [inline net/shorewall-lite/files/vardir]: Isn't /tmp removed after reboot? So I am missing the config files then. Or am I missing something.

Comment (lucize) [inline net/shorewall-lite/files/vardir]: ah, I think that the lite is not compiling it at every start

**Comment by wvdakker** [comment]
I have rebased the repos and removed the non-related commits. 

Running 5.1 forces you to install the Perl package. Even if you install the *lite version.

**Comment by wvdakker** [comment]
Had today contact with one of the guys from Shorewall. 5.1 lite can be build without Perl package.
I have build the package. Will test it later this weekend and upload it.

**Comment by hauke** [comment]
Please squash the commits together into one single commit

**Comment by wvdakker** [comment]
Made a single commit.

**Comment by hauke** [comment]
Please add the Signed-off-by line to the commit message again.

**Comment by wvdakker** [comment]
Done

**Pull request merged**